File: src/Matrix_Factorization/Matlab/id_map.m
function [u,v] = id_map(tr)
    u = zeros(1,730);
    v = zeros(1,6373);
    user = tr(:,1);
    movie = tr(:,2);
    user = sort(user);
    movie = sort(movie);
    u(1) = user(1);
    v(1) = movie(1);
    c1 = 2;
    for i=1:99999
        if user(i)~=user(i+1)
            u(c1) = user(i+1);
            c1 = c1 + 1;
        end
    end
    c2 = 2;
    for i=1:99999
        if movie(i)~=movie(i+1)
            v(c2) = movie(i+1);
            c2 = c2 + 1;
        end
    end
end

File: src/Matrix_Factorization/Python/id_map.py
from copy import *


def id_map(B):
    user = copy(B[:, 0])
    movie = copy(B[:, 1])
    user.sort()
    movie.sort()
    u = {user[0]: 0}
    v = {movie[0]: 0}

    c1 = 1
    for user_id in user:
        if user_id not in u:
            u.update({user_id: c1})
            c1 += 1

    c2 = 1
    for movie_id in movie:
        if movie_id not in v:
            v.update({movie_id: c2})
            c2 += 1

    return u, v